### PR TITLE
 Init partial network parameters from another saved model.

### LINKED ISSUE #2663: How to init partial network parameters from another saved model.
It's a useful and commonly function to use a trained model to init the network. But there is a bug by using [`from_tar`](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/parameters.py#L289) function. This interface is only suitable that all the parameter names are same in the initialized model and the network. If the users want to init partial network parameters through the initialized model, it will go wrong.  The bug is that the `Parameters` only contains the parameters in the initialized model and will omit some parameters in the defined network by using `from_tar`.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix #2663 

The usage example is:

```
para = paddle.parameters.create(cost)
para.init_from_tar('init_params.tar.gz')
```

### CODE REVIEW

File: python/paddle/v2/parameters.py [hunk 1]
```diff
@@ -187,7 +184,7 @@ def __setitem__(self, key, value):
                              (shape, value.shape))
 
         if len(self.__gradient_machines__) == 0:
-            self.__tmp_params__.append((key, value))
```
Comment (qingqing01) [inline python/paddle/v2/parameters.py]: The `append` could lead the repeated key in `self.__tmp_params__`. This is a bug.

File: python/paddle/v2/parameters.py [hunk 2]
```diff
@@ -302,6 +299,12 @@ def from_tar(f):
             params.deserialize(param_name, f)
         return params
 
+    def init_from_tar(self, f):
```
Comment (reyoung) [inline python/paddle/v2/parameters.py]: Because `from_tar` is a static method. Maybe `Parameters.from_tar` is better.

Comment (qingqing01) [inline python/paddle/v2/parameters.py]: Done.

**Comment by qingqing01** [comment]
@pkuyym add comments.

**Comment by NHZlX** [comment]
The usage example should be :
```
para = paddle.parameters.create(cost)
with gzip.open('init_params.tar.gz') as f:
     para.init_from_tar(f)
```

**Pull request merged**